engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala

```diff
@@ -102,3 +111,11 @@ class FlinkProcessCompiler(creator: ProcessConfigCreator,
     }
   }
 }
+
+//The logic of listeners invocation is a bit tricky for CustomNodes/Sinks (see Interpreter.interpretNode)
+//Metric listeners have to initialize counters in appropriate parts...
+private[process] case class UsedNodes(nodes: Iterable[NodeData], nextParts: Iterable[String])
```

Comment:
existence of `nodes` are pretty obvious here, but I don't get why `nextParts` are also necessary. can you write a few words about that in comment?

Reply:
I added longer comment...